cql3/statements/modification_statement.cc

```diff
@@ -259,23 +264,26 @@ modification_statement::do_execute(query_processor& qp, service::query_state& qs
         return execute_with_condition(qp, qs, options);
     }
 
-    return execute_without_condition(qp, qs, options).then([] {
+    return execute_without_condition(qp, qs, options).then([] (coordinator_result<> res) {
+        if (res.has_error()) {
+            return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
+                    seastar::make_shared<cql_transport::messages::result_message::exception>(std::move(res).assume_error()));
+        }
         return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
                 ::shared_ptr<cql_transport::messages::result_message>{});
     });
 }
 
-future<>
+future<coordinator_result<>>
 modification_statement::execute_without_condition(query_processor& qp, service::query_state& qs, const query_options& options) const {
     auto cl = options.get_consistency();
     auto timeout = db::timeout_clock::now() + get_timeout(qs.get_client_state(), options);
     return get_mutations(qp, options, timeout, false, options.get_timestamp(qs), qs).then([this, cl, timeout, &qp, &qs] (auto mutations) {
         if (mutations.empty()) {
-            return now();
+            return make_ready_future<coordinator_result<>>(bo::success());
```

Comment:
Again, is this bo::success() really needed? If it's not (nothing, or {}, can be used instead), you won't need to define this ugly namespace "bo" above.

Reply:
As explained in another discussion, `result<>` has its own default constructor explicitly deleted. For results holding `void` using `success()` seems to be the easiest way to initialize a result with value.